tools/cellprofiler/image_math.py

```diff
@@ -0,0 +1,104 @@
+import json
+import sys
+import os
+import argparse
+from cp_common_functions import *
+
+MODULE_NAME = "ImageMath"
+OUTPUT_FILENAME = "output.cppipe"
+
+operator_map = {
+    "add": "Add",
+    "subtract": "Subtract",
+    "multiply": "Multiply",
+    "divide": "Divide",
+    "average": "Average",
+    "minimum": "Minimum",
+    "maximum": "Maximum",
+    "invert": "Invert",
+    "log_2": "Log transform (base 2)",
+    "log_legacy": "Log transform (legacy)",
+    "and": "And",
+    "or": "Or",
+    "not": "Not",
+    "equals": "Equals"
+}
+
+
+def build_main_block(input_params):
+    """Creates the main block of the CP pipeline with the parameters that don't depend on conditional choices"""
+    operation = operator_map[get_json_value(
+        input_params, 'operation.operation')]
+    result = INDENTATION.join(
+        [f"{INDENTATION}Operation:{operation}\n",
+         f"Raise the power of the result by:{get_json_value(input_params,'operation.op_results.raise_the_power_of_the_result_by')}\n",
+         f"Multiply the result by:{get_json_value(input_params,'operation.op_results.multiply_the_result_by')}\n",
+         f"Add to result:{get_json_value(input_params,'operation.op_results.add_to_result')}\n",
+         f"Set values less than 0 equal to 0?:{get_json_value(input_params,'operation.op_results.set_values_less_than_0_equal_to_0')}\n",
+         f"Set values greater than 1 equal to 1?:{get_json_value(input_params,'operation.op_results.set_values_greater_than_1_equal_to_1')}\n",
+         f"Ignore the image masks?:{get_json_value(input_params,'ignore_the_image_masks')}\n",
+         f"Name the output image:{get_json_value(input_params,'name_output_image')}"
+         ])
+    return(result)
+
+
+def build_variable_block(inputs_galaxy):
+    result = ""
+    first_image_block = build_first_image_block(
+        get_json_value(inputs_galaxy, 'operation.first_image'))
+    result = "\n".join([result, first_image_block])
+    second_image_block = build_second_image_block(
+        get_json_value(inputs_galaxy, 'operation.second_image'))
+    result = "\n".join([result, second_image_block])
```

Comment:
or why not simply summing everything up only once and safe the first result step?
Please keep in mind you can do `a += b`